plugins/modules/aci_cloud_aws_provider.py

```diff
@@ -0,0 +1,309 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, nkatarmal-crest <nirav.katarmal@crestdatasys.com>
+# Copyright: (c) 2021, Shreyas Srish <ssrish@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: aci_cloud_aws_provider
+short_description: Manage Cloud AWS Provider (cloud:AwsProvider)
+description:
+- Manage AWS provider on Cisco Cloud ACI.
+author:
+- Devarshi Shah (@devarshishah3)
+options:
+  access_key_id:
+    description:
+    - Cloud Access Key ID.
+    type: str
+  account_id:
+    description:
+    - AWS Account ID.
+    type: str
+  email:
+    description:
+    - Account Email Address.
+    type: str
+  http_proxy:
+    description:
+    - Http Proxy to connect to cloud provider.
+    type: str
+  is_account_in_org:
+    description:
+    - Is Account in Organization.
+    type: bool
+  is_trusted:
+    description:
+    - Trusted Tenant
+    type: bool
+  provider_id:
+    description:
+    - AWS provider id
+    type: str
+  region:
+    description:
+    - AWS Region.
+    type: str
```

Comment:
What is the region used for?

Reply:
It is part of cloud resources of the tenant. I was unable to change it via API because the tenant in question was already associated with regions after creation. I left it just in case once could use it when no regions are associated upon creation..